### ISSUE TITLE
Root path checking is incorrect

### ISSUE THREAD

**Issue by chi-lee** [patriksimek/vm2#281]
If I create a vm with the config containing "require.root" equal "/test", it would accept modules from '/test-env/**' as well because the module path is compared to the root using String.prototype.startsWith().

### Expected Behavior
Should throw ```VMError(`Module '${moduleName}' is not allowed to be required. The path is outside the border!`, 'EDENIED')```

### Current Behavior
The module is loaded.

### Possible Solution
Change the comparison from using String.prototype.startsWith() to real path checking.

### Steps to Reproduce
1. Create folders called 'test', and 'test-env'.
2. Run 'npm install lodash' inside test-env
3. Install vm2 in 'test' and create a script of creating a vm with the config containing "require: { root: '.' }"
4. Run the script "require('lodash')" using the vm.

**Comment by XmiliaH** [comment]
I'm not sure if this is intended or not. You could easily just set require.root to '/test/' to get the desired result.

**Comment by chi-lee** [comment]
I agree that I can just set require.root to '/test/' but the behavior surprised me and I'm sure there will be other people facing the same issue somehow. Let's ask @patriksimek if this is intended or not. I could help change that part of code and document it afterwards if you guys like.

**Comment by XmiliaH** [comment]
This part will change because it used the sandbox String.prototype.startsWith method which could be overwritten by scripts. So I would just add this part there too. See https://github.com/patriksimek/vm2/pull/277/files#diff-2e9fddd3c1dd98f4bc2e2a3e93205d53R376.
Asking patriksimek might take some time.
